ql/src/java/org/apache/hadoop/hive/ql/exec/FetchOperator.java

```diff
@@ -227,7 +227,11 @@ public static InputFormat getInputFormatFromCache(
     if (format == null) {
       try {
         format = ReflectionUtil.newInstance(inputFormatClass, conf);
-        inputFormats.put(inputFormatClass.getName(), format);
+        // HBase input formats are not thread safe today. See HIVE-8808.
+        String inputFormatName = inputFormatClass.getName().toLowerCase();
+        if (!inputFormatName.contains("hbase")) {
```

Comment:
Just a question: Refer to HIVE-8808, other inputformat, e.g jdbc & kudu & kafka will be stateful?  and can these inputformat also have the same issue?

Reply:
Yes, those input formats can also have the same issue if those are stateful or not thread safe. Depends on their implementation.

Reply:
Looks like the JdbcInputFormat is stateful, but we can make it stateless, we can fix it in the follow-up jira.
https://github.com/apache/hive/blob/a406d6d4417277e45b93f1733bed5201afdee29b/jdbc-handler/src/main/java/org/apache/hive/storage/jdbc/JdbcInputFormat.java#L46

Reply:
Sure, we can fix in follow-up jira if other input format has similar issue.